core/lib/zksync_core/src/basic_witness_input_producer/mod.rs

```diff
@@ -0,0 +1,203 @@
+                tracing::debug!("Finished execution of miniblock: {miniblock:?}");
+            }
+            vm.finish_batch();
+            tracing::info!("Finished execution of l1_batch: {:?}", job.l1_batch_number);
+
+            metrics::histogram!(
+                "basic_witness_input_producer.input_producer_time",
+                started_at.elapsed(),
+            );
+            tracing::info!(
+                "BasicWitnessInputProducer took {:?} for L1BatchNumber {}",
+                started_at.elapsed(),
+                job.l1_batch_number.0
+            );
+
+            let witness_block_state = (*storage_view).borrow().witness_block_state();
+            Ok(witness_block_state)
+        })
+        .await?
+    }
+}
+
+#[async_trait]
+impl JobProcessor for BasicWitnessInputProducer {
+    type Job = BasicWitnessInputProducerJob;
+    type JobId = L1BatchNumber;
+    type JobArtifacts = WitnessBlockState;
+    const SERVICE_NAME: &'static str = "";
+
+    async fn get_next_job(&self) -> anyhow::Result<Option<(Self::JobId, Self::Job)>> {
+        let mut connection = self
+            .connection_pool
+            .access_storage()
+            .await
+            .expect("couldn't get a connection from the pool");
+        let l1_batch_to_process = connection
+            .basic_witness_input_producer_dal()
+            .get_next_basic_witness_input_producer_job()
+            .await;
+        match l1_batch_to_process {
+            Some(number) => Ok(Some((
+                number,
+                BasicWitnessInputProducerJob {
+                    l1_batch_number: number,
+                },
+            ))),
+            None => Ok(None),
+        }
+    }
+
+    async fn save_failure(&self, job_id: Self::JobId, started_at: Instant, error: String) {
+        let attempts = self
+            .connection_pool
+            .access_storage()
+            .await
+            .unwrap()
+            .basic_witness_input_producer_dal()
+            .mark_job_as_failed(job_id, started_at, error)
+            .await
+            .expect("didn't receive number of attempts from database");
+        tracing::warn!(
+            "Failed to process job: {:?}, attempts: {}",
+            job_id,
+            attempts
+        );
+    }
+
+    async fn process_job(
+        &self,
+        job: Self::Job,
+        started_at: Instant,
+    ) -> JoinHandle<anyhow::Result<Self::JobArtifacts>> {
+        tokio::spawn(Self::process_job_impl(
+            job,
+            started_at,
+            self.connection_pool.clone(),
+            self.validation_computational_gas_limit,
+            self.l2_chain_id,
+        ))
+    }
+
+    async fn save_result(
+        &self,
+        job_id: Self::JobId,
+        started_at: Instant,
+        artifacts: Self::JobArtifacts,
+    ) -> anyhow::Result<()> {
+        let upload_started_at = Instant::now();
+        let _object_path = self.object_store.put(job_id, &artifacts).await?;
```

Comment:
shall we persist the `_object_path` retrieved from gcs into db?

Reply:
yes, if for nothing else than consistency with other components.

Reply:
I see, if we intend to not populate the db with blob_url, then the  blob_url column isn't very useful since it would be always empty.
For debugging purposes it very handy to have the `blob_url` present in the db.